File: challenges/a-challenge-1/main.py
"""Write a function that takes a string of characters and returns a dictionary with each word in the string as the key and its length as the value. For example, if the function receives the string "Hello everyone" it should return {'Hello': 4, 'to': 1, 'everyone': 5}.
"""


def word_lengths(string):
    # Split the string into a list of words
    words = string.split()

    # Create the result dictionary with a dictionary comprehension
    result = {word: len(word) for word in words}

    return result

File: challenges/a-challenge-1/main.cpp
#include <iostream>
#include <string>
#include <unordered_map>

std::unordered_map<std::string, int> word_lengths(const std::string &string)
{
    // Split the string into a vector of words
    std::vector<std::string> words;
    std::istringstream iss(string);
    for (std::string word; iss >> word;)
    {
        words.push_back(word);
    }

    // Create the result unordered_map with an unordered_map comprehension
    std::unordered_map<std::string, int> result;
    for (const auto &word : words)
    {
        result[word] = word.length();
    }

    return result;
}

int main()
{
    std::string string = "Hello everyone";
    auto result = word_lengths(string);
    for (const auto &[word, length] : result)
    {
        std::cout << word << ": " << length << std::endl;
    }
    return 0;
}
